src/tsdb.c

```diff
@@ -325,6 +424,21 @@ ChunkResult SeriesIteratorGetNext(SeriesIterator *iterator, Sample *currentSampl
     return CR_OK;
 }
 
+int SeriesUpdateLastSample(Series *series, Sample *sample) {
```

Comment:
@ashtul it looks like this function doesn't update the last sample, instead it retrieves it